src/magma/vk/textures/TextureManager.cpp

```diff
@@ -0,0 +1,213 @@
+/**
+ * @file TextureManager.cpp
+ * @brief Create and control textures
+ * @version 0.1
+ * @date 2021-02-12
+ */
+#include "magma/vk/textures/TextureManager.h"
+
+#include <stdexcept>
+#include <vulkan/vulkan_core.h>
+
+#include "magma/app/log.hpp"
+#include "magma/vk/logicalDevice.h"
+#include "magma/vk/vulkan_common.h"
+
+TextureManager::TextureManager(LogicalDevice &device) 
+        : _device(device),
+        _commandBuffers(device.handler(), device.getGraphicsCmdPool(), 1)
+{}
+
+TextureManager::~TextureManager() {
+    _commandBuffers.freeCmdBuf(_device.handler(), _device.getGraphicsCmdPool());
+    if (_textures.size() > 0) {
+        LOG_WARNING(_textures.size(), " textures not been removed");
+        while(_textures.size() > 0) {
+            deleteTexture(_textures.begin()->second);
+        }
+    }
+}
+
+bool TextureManager::textureExist(const std::string &name) const {
+    return _textures.find(name) != _textures.end();
+}
+
+Texture& TextureManager::getTexture(const std::string &name) {
+    if (textureExist(name)) {
+        return _textures.at(name);
+    } else {
+        throw std::invalid_argument("TextureManager::getTexture::texture not exist");
```

Comment:
last "::" looks confusing, because there is no method of member, called:
```
TextureManager::getTexture::texture
```

Reply:
Done